packages/domain/lib/usecase/get_movie_anticipated_response_usecase.dart

```diff
@@ -0,0 +1,40 @@
+import '../utils/const.dart';
+import '../model/movie_anticipated_response.dart';
+import '../repository/movie_repository.dart';
+import 'usecase.dart';
+
+class GetMovieAnticipatedResponseUseCase
+    implements UseCase<Future<List<MovieAnticipatedResponse>>> {
+  final NetworkRepository _repository;
+
+  GetMovieAnticipatedResponseUseCase(this._repository);
+
+  @override
+  Future<List<MovieAnticipatedResponse>> call() async {
+    final List<MovieAnticipatedResponse> jsonMovies = [];
+    final response =
+        await _repository.getMovieData(apiPath: C.apiAnticipatedPath);
+    if (int.parse(response.headers['x-pagination-page-count'][0]) >= 5) {
+      const int itemCount = 50;
+      await _addToJsonMovies(itemCount, jsonMovies);
```

Comment:
1) naming - _addToJsonMovies doesn't say anything about api call to get more films
2) I wrote it higher - here u should take elements from the second page, not from the first